From 47c67fd7a59f521699c9c8d98910167455edb40d Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Mon, 14 Jul 2014 18:58:35 -0400
Subject: [idle] intel_idle: fix IVT idle state table setting

Message-id: <1405364315-27440-8-git-send-email-prarit@redhat.com>
Patchwork-id: 85801
O-Subject: [RHEL7.1 PATCH BZ 1119374 7/7] intel_idle: fix IVT idle state table setting
Bugzilla: 1119374
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Rik van Riel <riel@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1119374

commit d27dca4217eb4cbdc3d33ad8c07799dd184873b9
Author: Christoph Jaeger <christophjaeger@linux.com>
Date:   Sat Apr 12 19:57:30 2014 +0200

    intel_idle: fix IVT idle state table setting

    Ivy Town idle state table will not be set as intended. Fix it.

    Picked up by Coverity - CID 1201420/1201421.

    Fixes: 0138d8f075 ("intel_idle: fine-tune IVT residency targets")
    Signed-off-by: Christoph Jaeger <christophjaeger@linux.com>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/idle/intel_idle.c b/drivers/idle/intel_idle.c
index cf78dc1..5d450ec 100644
--- a/drivers/idle/intel_idle.c
+++ b/drivers/idle/intel_idle.c
@@ -672,9 +672,10 @@ void intel_idle_state_table_update(void)
    if (package_num + 1 > num_sockets) {
     num_sockets = package_num + 1;
 
-    if (num_sockets > 4)
+    if (num_sockets > 4) {
      cpuidle_state_table = ivt_cstates_8s;
      return;
+    }
    }
   }
 
-- 
1.7.1